src/veh_type.cpp

```diff
@@ -326,8 +365,65 @@ void vpart_info::finalize()
 
 void vpart_info::check()
 {
-    for( auto &part_ptr : vehicle_part_int_types ) {
-        auto &part = *part_ptr;
+    for( auto &vp : vehicle_part_types ) {
+        auto &part = vp.second;
+
+        // handle legacy parts without requirement data
+        if( part.legacy ) {
+            part.install_reqs.components = { { { { part.item, 1 } } } };
+
+            part.install_skills.emplace( skill_mechanics, part.difficulty );
+            part.removal_skills.emplace( skill_mechanics, part.difficulty );
+
+            if( part.has_flag( "TOOL_WRENCH" ) || part.has_flag( "WHEEL" ) ) {
+                part.install_reqs.qualities = { { { quality_id( "WRENCH" ), 1, 1 } } };
+                part.removal_reqs.qualities = { { { quality_id( "WRENCH" ), 1, 1 } } };
+            } else if( part.has_flag( "TOOL_SCREWDRIVER" ) ) {
+                part.install_reqs.qualities = { { { { quality_id( "SCREW" ), 1, 1 } } } };
+                part.removal_reqs.qualities = { { { { quality_id( "SCREW" ), 1, 1 } } } };
+            } else if( part.has_flag( "NAILABLE" ) ) {
+                part.install_reqs.qualities = { { { { quality_id( "HAMMER" ), 1, 1 } } } };
+                part.removal_reqs.qualities = { { { { quality_id( "HAMMER" ), 1, 1 } } } };
+                part.install_reqs.components.push_back( { { { "nail", 20 } } } );
+            } else if( part.has_flag( "TOOL_NONE" ) ) {
+                // intentional no-op as we require nothing
+            } else {
+                part.install_reqs.qualities = { { { { quality_id( "GLARE" ), 1, 2 } } } };
+                part.install_reqs.tools.push_back( { { { "welder", 50 }, { "welder_crude", 75 }, { "oxy_torch", 25 } } } );
```

Comment:
Oxytorch charges should be 10.

Install requirements should also include wrench.
Removal requirements should be fine metal sawing and wrench.